homeassistant/components/screenlogic/climate.py

```diff
@@ -0,0 +1,214 @@
+"""Support for a ScreenLogic heating device."""
+import logging
+
+from screenlogicpy.const import HEAT_MODE
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_PRESET_MODE,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from . import ScreenlogicEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_FEATURES = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+SUPPORTED_MODES = [HVAC_MODE_OFF, HVAC_MODE_HEAT]
+
+SUPPORTED_PRESETS = [
+    HEAT_MODE.SOLAR,
+    HEAT_MODE.SOLAR_PREFERRED,
+    HEAT_MODE.HEATER,
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    entities = []
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    coordinator = data["coordinator"]
+
+    for body in data["devices"]["body"]:
+        entities.append(ScreenLogicClimate(coordinator, body))
+    async_add_entities(entities)
+
+
+class ScreenLogicClimate(ScreenlogicEntity, ClimateEntity, RestoreEntity):
+    """Represents a ScreenLogic climate entity."""
+
+    def __init__(self, coordinator, body):
+        """Initialize a ScreenLogic climate entity."""
+        super().__init__(coordinator, body)
+        self._configured_heat_modes = []
+        # Is solar listed as available equipment?
+        if self.coordinator.data["config"]["equipment_flags"] & 0x1:
```

Comment:
It would be cleaner to make `0x1` a constant imported from `screenlogicpy`

Reply:
Commit https://github.com/home-assistant/core/pull/48136/commits/2d488e2a1747d24006a99a3b5f374c69f73748c7